src/api/useFetch.js

```diff
@@ -19,6 +19,65 @@ import { FeatureRegistrationContext } from '../App'
 import { registerCallbackHandler } from '../feature'
 
 
+const useFetch = (
+  route,
+  params,
+  errorMsg,
+) => {
+  const [res, setRes] = React.useState()
+  const [isLoading, setIsLoading] = React.useState(true)
+  const [isError, setIsError] = React.useState()
+  const [error, setError] = React.useState()
```

Comment:
do we need both `IsError` and `error`?
might it make sense to have a `state` rather than `isLoading` + `isError`

Reply:
> do we need both IsError and error?

we should store the actual error msg as well, it is incorporated into snackbar notification (for most requests, at least)

> might it make sense to have a state rather than isLoading + isError

I like the state idea, it is more generic and could be easily extended if needed

Reply:
of course we should store the error-message. however, is it not sufficient to just store the message and assume `null` / `undefined` is equivalent to "no error"?